src/app/app.component.ts

```diff
@@ -1,24 +1,39 @@
-import { Actions, JsonSchema, UISchemaElement } from '@jsonforms/core';
-import { ChangeDetectionStrategy, ChangeDetectorRef, Component } from '@angular/core';
+import { Actions } from '@jsonforms/core';
+import {
+  Component,
+  Input,
+  OnChanges,
+  SimpleChanges,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+} from '@angular/core';
 
 import { JsonFormsAngularService } from '@jsonforms/angular';
 
 @Component({
-  selector: 'app-root',
+  selector: 'ng-jsonforms',
   template: `<jsonforms-outlet></jsonforms-outlet>`,
-  changeDetection: ChangeDetectionStrategy.OnPush
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class AppComponent {
+export class AppComponent implements OnChanges {
+  @Input() uischema: string;
+  @Input() schema: string;
+  @Input() data: string;
 
-  constructor(private jsonformsService: JsonFormsAngularService, private cdr: ChangeDetectorRef) {
-  }
+  constructor(
+    private jsonformsService: JsonFormsAngularService,
+    private cdr: ChangeDetectorRef
+  ) {}
 
-  setJsonFormsInput(
-    schema: JsonSchema,
-    uiSchema?: UISchemaElement,
-    data: any = {}
-  ) {
-    this.jsonformsService.updateCore(Actions.init(data, schema, uiSchema));
+  ngOnChanges(changes: SimpleChanges): void {
+    const dataObject = this.data ? JSON.parse(this.data) : undefined;
```

Comment:
I think we should default data to `{}`
```suggestion
    const dataObject = this.data ? JSON.parse(this.data) : {};
```

Reply:
This is not necessary anymore with the latest version of jsonforms :smile:

Reply:
... but the project still uses the old version. Pushing a commit to update the version to 2.4.0